### PR TITLE
GIX-1987: Load Tokens User Data

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Render the tokens of the logged in user.

In this PR, we load the balances of SNS and ckBTC projects in the tokens route.

I used the same way that they are loaded in accounts. It has been working well.

# Changes

* Load SNS and ckBTC balances in tokens route.

# Tests

* Add a test to Tokens route that checks balances of the user when logged in.
* Add various methods to tokens table related page objects.

# Todos

- [ ] Add entry to changelog (if necessary).

Not yet necessary.

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3758** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/3758?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/3758?utm_source=stack-comment).

**Comment by lmuntaner** [comment]
Same UI, but loading data:

**Comment by lmuntaner** [comment]
@dskloetd @mstrasinskis please review

**Comment by dskloetd** [comment]
> I used the same way that they are loaded in accounts. It has been working well.

Would it make sense to share the code instead of duplicating it?

File: frontend/src/tests/page-objects/TokensTableRow.page-object.ts [hunk 1]
```diff
@@ -20,6 +20,12 @@ export class TokensTableRowPo extends BasePageObject {
     return this.getText("token-value-label");
   }
 
+  async getData(): Promise<Record<string, string>> {
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/TokensTableRow.page-object.ts:23]: Why `Record<string, string>` and not `{ projectName: string; balance: string }`?

File: frontend/src/tests/routes/app/tokens/page.spec.ts [hunk 2]
```diff
@@ -34,6 +42,33 @@ describe("Tokens route", () => {
...
+      );
+      vi.spyOn(snsLedgerApi, "getSnsAccounts").mockImplementation(
+        async ({ rootCanisterId }) => {
+          if (rootCanisterId.toText() === rootCanisterId1.toText()) {
+            return [mockSnsMainAccount];
+          }
+          return [
+            {
+              ...mockSnsMainAccount,
+              balanceE8s: 314000000n,
```
Comment (dskloetd) [inline frontend/src/tests/routes/app/tokens/page.spec.ts:56]: I'm missing something. Can you explain what's going on here?
Why do we set the balance except for a specific canisterId?
Do we want to set a different balance for `rootCanisterId1`?
Why should the balance (but nothing else) depend on the canisterId?

File: frontend/src/tests/page-objects/TokensTableRow.page-object.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/tests/page-objects/TokensTableRow.page-object.ts:23]: I wanted it to be generic...

But it makes sense to be specific actually.

File: frontend/src/tests/routes/app/tokens/page.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/routes/app/tokens/page.spec.ts:56]: Yes, I want to set a different balance based on rootCanisterId.

the account is:
```
  identifier: encodeIcrcAccount({
    owner: mockPrincipal,
  }),
  balanceE8s: 890156712340000n,
  principal: mockPrincipal,
  type: "main",
```

Nothing is specific to the SNS. The difference is more where it's located, than in the data itself.

I want a difference balance, just to test that it works with multiple SNSes.

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Comment by lmuntaner** [comment]
> Would it make sense to share the code instead of duplicating it?

I will be removing the code there as part of the clean up once the feature flag is enabled in prod.

File: frontend/src/tests/routes/app/tokens/page.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/routes/app/tokens/page.spec.ts]: Yes, good point. I did the same for ckbtc and icp also.

File: frontend/src/tests/routes/app/tokens/page.spec.ts [hunk 3]
```diff
@@ -43,6 +51,33 @@ describe("Tokens route", () => {
...
+            },
+          ];
+        }
+      );
+      vi.spyOn(snsLedgerApi, "getSnsToken").mockImplementation(
+        async ({ rootCanisterId }) => {
+          if (rootCanisterId.toText() === rootCanisterId1.toText()) {
+            return mockSnsToken;
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/routes/app/tokens/page.spec.ts]: done

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Review by dskloetd — APPROVED** [review]

**Pull request merged**